src/lib/reasoners/bitv_rel.ml

```diff
@@ -395,23 +421,39 @@ let add_eqs =
 
    Iterate until fixpoint is reached. *)
 let propagate =
-  let rec propagate changed bcs dom =
-    match Constraints.next_pending bcs with
-    | { value; explanation = ex }, bcs ->
+  let touch bcs queue r =
+    QC.push queue (Structural r);
+    Constraints.iter_parents (fun c -> QC.push queue (Constraint c)) r bcs
```

Comment:
Not really, I don't think there is a particularly significant order between constraints here. I did put the `Structural` one first because it is the strongest for the bit-vector domain, in particular when a leaf is updated (any bit known in the leaf will be known in the semantic values that contain the leaves) and so the most likely to trigger future propagations, so putting it first probably is slightly less likely to cause duplicate propagations later, but I am not sure it actually matters.